packages/transformer/src/IModelTransformer.ts

```diff
@@ -411,18 +414,144 @@ export class IModelTransformer extends IModelExportHandler {
+    opts?: { getJsonProperties?: boolean; preserveVersionReceived?: boolean }
+  ): void {
+    const sql = `
+      SELECT ECInstanceId, Version
+      ${opts?.getJsonProperties ? ", JsonProperties" : ""}
+      FROM ${ExternalSourceAspect.classFullName}
+      WHERE Element.Id=:elementId
+        AND Scope.Id=:scopeId
+        AND Kind=:kind
+        AND Identifier=:identifier
+      LIMIT 1
+    `;
+    dbToQuery.withPreparedStatement(sql, (statement: ECSqlStatement) => {
+      statement.bindId("elementId", aspectProps.element.id);
+      if (aspectProps.scope === undefined) return; // return instead of binding an invalid id
+      statement.bindId("scopeId", aspectProps.scope.id);
+      statement.bindString("kind", aspectProps.kind);
+      statement.bindString("identifier", aspectProps.identifier);
+      if (DbResult.BE_SQLITE_ROW !== statement.step()) return;
+      aspectProps.id = statement.getValue(0).getId();
+      aspectProps.version = opts?.preserveVersionReceived
+        ? aspectProps.version
+        : statement.getValue(1).getString();
+      aspectProps.jsonProperties = opts?.getJsonProperties
+        ? JSON.parse(statement.getValue(2).getString())
+        : undefined;
+    });
   }
 
-  private get _isForwardSynchronization() {
-    return this._isSynchronization && !this._options.isReverseSynchronization;
+  /**
+   * Determines the sync direction "forward" or "reverse" of a given sourceDb and targetDb by looking for the scoping ESA.
+   * If the sourceDb's iModelId is found as the identifier of the expected scoping ESA in the targetDb, then it is a forward synchronization.
+   * If the targetDb's iModelId is found as the identifier of the expected scoping ESA in the sourceDb, then it is a reverse synchronization.
+   * @throws if no scoping ESA can be found in either the sourceDb or targetDb which describes a master branch relationship between the two databases.
+   * @param sourceDb
+   * @param targetDb
+   * @param targetScopeElementId the element Id defined on the ExternalSourceAspect's scope. @see [[ExternalSourceAspectProps]]
+   * @returns "forward" or "reverse"
+   */
+  public static determineSyncType(
+    sourceDb: IModelDb,
+    targetDb: IModelDb,
+    targetScopeElementId: Id64String
+  ): "forward" | "reverse" {
+    const aspectProps = {
+      id: undefined as string | undefined,
+      version: undefined as string | undefined,
+      classFullName: ExternalSourceAspect.classFullName,
+      element: {
+        id: targetScopeElementId,
+        relClassName: ElementOwnsExternalSourceAspects.classFullName,
+      },
+      scope: { id: IModel.rootSubjectId }, // the root Subject scopes scope elements
+      identifier: sourceDb.iModelId,
+      kind: ExternalSourceAspect.Kind.Scope,
+      jsonProperties: undefined as TargetScopeProvenanceJsonProps | undefined,
+    };
+    // First assume that the targetDb is the provenanceDb/branch (meaning holds the provenance which points back to master)
```

Comment:
```suggestion
    // First check if the targetDb is the branch 
```

Reply:
I think I'd prefer to keep the lengthier comment, but I did change assume to 'check if'. I just think it gives more context to someone looking at this repo for the first time

Reply:
I would personally rather something like:
```
//First check if the target is the branch (branch is the @see provenanceDb)
```
I prefer we link back to centralized doc concepts as much as possible

Reply:
Yeah thats better, changed it.